syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/sql/SyndeoScalarSqlTransformSchemaTransformProvider.java

```diff
@@ -0,0 +1,87 @@
+ * the License.
+ */
+package com.google.cloud.syndeo.transforms.sql;
+
+import com.google.auto.service.AutoService;
+import com.google.auto.value.AutoValue;
+import java.util.Collections;
+import java.util.List;
+import org.apache.beam.sdk.extensions.sql.expansion.SqlTransformSchemaTransformProvider;
+import org.apache.beam.sdk.schemas.AutoValueSchema;
+import org.apache.beam.sdk.schemas.annotations.DefaultSchema;
+import org.apache.beam.sdk.schemas.logicaltypes.EnumerationType;
+import org.apache.beam.sdk.schemas.transforms.SchemaTransform;
+import org.apache.beam.sdk.schemas.transforms.SchemaTransformProvider;
+import org.apache.beam.sdk.schemas.transforms.TypedSchemaTransformProvider;
+import org.apache.beam.sdk.values.Row;
+import org.checkerframework.checker.initialization.qual.Initialized;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.UnknownKeyFor;
+
+@AutoService(SchemaTransformProvider.class)
+public class SyndeoScalarSqlTransformSchemaTransformProvider
+    extends TypedSchemaTransformProvider<
+        SyndeoScalarSqlTransformSchemaTransformProvider
+            .SyndeoScalarSqlTransformSchemaTransformConfiguration> {
+
+  @Override
+  protected @UnknownKeyFor @NonNull @Initialized Class<
+          SyndeoScalarSqlTransformSchemaTransformConfiguration>
+      configurationClass() {
+    return SyndeoScalarSqlTransformSchemaTransformConfiguration.class;
+  }
+
+  @Override
+  protected @UnknownKeyFor @NonNull @Initialized SchemaTransform from(
+      SyndeoScalarSqlTransformSchemaTransformConfiguration configuration) {
+    SqlTransformSchemaTransformProvider provider = new SqlTransformSchemaTransformProvider();
+
+    EnumerationType enumtype =
+        ((EnumerationType)
+            provider.configurationSchema().getField("dialect").getType().getLogicalType());
+
+    String query =
+        "SELECT " + String.join(", ", configuration.getColumnExpressions()) + " FROM input";
+    return provider.from(
+        Row.withSchema(provider.configurationSchema())
+            .withFieldValue("query", query)
+            .withFieldValue("dialect", enumtype.valueOf("zetasql"))
+            .build());
+  }
+
+  @Override
+  public @UnknownKeyFor @NonNull @Initialized String identifier() {
+    return "syndeo:schematransform:com.google.cloud:sql_scalar_transform:v1";
+  }
+
+  @Override
+  public @UnknownKeyFor @NonNull @Initialized List<@UnknownKeyFor @NonNull @Initialized String>
+      inputCollectionNames() {
+    return Collections.singletonList("input");
+  }
+
+  @Override
+  public @UnknownKeyFor @NonNull @Initialized List<@UnknownKeyFor @NonNull @Initialized String>
+      outputCollectionNames() {
+    return List.of("output", "errors");
+  }
+
+  @DefaultSchema(AutoValueSchema.class)
+  @AutoValue
+  public abstract static class SyndeoScalarSqlTransformSchemaTransformConfiguration {
+    public abstract List<String> getColumnExpressions();
```

Comment:
Maybe rename this to something that reflects that this is for the `SELECT` statement, like `getSelectColumnExpressions` or, in case this opens up to more than just columns, `getSelectExpressions`. That way, this is clearly separated from any other expression lists that we support (e.g. if we start supporting filters).

Reply:
done